brayns/engine/Scene.cpp

```diff
@@ -20,522 +21,119 @@
-            {0.5f - lampInfo.x, lampInfo.y, 0.5f + lampInfo.z}};
-        auto &triangleMesh = model->getTriangleMeshes()[materialId];
-        for (size_t i = 0; i < 4; ++i)
-            triangleMesh.vertices.push_back(lampPositions[i]);
-        triangleMesh.indices.push_back(Vector3i(2, 1, 0));
-        triangleMesh.indices.push_back(Vector3i(0, 3, 2));
-    }
-
-    addModel(std::make_shared<ModelDescriptor>(std::move(model), "DefaultScene"));
-}
-
-bool Scene::empty() const
-{
-    auto lock = acquireReadAccess();
-    for (auto modelDescriptor : _modelDescriptors)
-        if (!modelDescriptor->getModel().empty())
-            return false;
-    return true;
-}
-
-size_t Scene::addClipPlane(const Plane &plane)
-{
-    auto clipPlane = std::make_shared<ClipPlane>(plane);
-    clipPlane->onModified([&](const BaseObject &) { markModified(false); });
-    _clipPlanes.emplace_back(std::move(clipPlane));
-    markModified();
-    return _clipPlanes.back()->getID();
-}
-
-ClipPlanePtr Scene::getClipPlane(const size_t id) const
-{
-    return _find(_clipPlanes, id);
-}
-
-void Scene::removeClipPlane(const size_t id)
-{
-    if (_remove(_clipPlanes, id))
-        markModified();
-}
-
-const std::vector<ClipPlanePtr> &Scene::getClipPlanes() const
-{
-    return _clipPlanes;
-}
-
-size_t Scene::getSizeInBytes() const
-{
-    auto lock = acquireReadAccess();
-    size_t sizeInBytes = 0;
-    for (auto modelDescriptor : _modelDescriptors)
-        sizeInBytes += modelDescriptor->getModel().getSizeInBytes();
-    return sizeInBytes;
-}
-
-size_t Scene::getNumModels() const
-{
-    auto lock = acquireReadAccess();
-    return _modelDescriptors.size();
-}
-
-void Scene::setMaterialsColorMap(MaterialsColorMap colorMap)
-{
-    {
-        auto lock = acquireReadAccess();
-        for (auto modelDescriptors : _modelDescriptors)
-            modelDescriptors->getModel().setMaterialsColorMap(colorMap);
-    }
-    markModified();
-}
-
-MaterialPtr Scene::getBackgroundMaterial() const
-{
-    return _backgroundMaterial;
-}
-
-TransferFunction &Scene::getTransferFunction()
-{
-    return _transferFunction;
-}
-
-const TransferFunction &Scene::getTransferFunction() const
-{
-    return _transferFunction;
-}
-
-void Scene::addModels(std::vector<ModelDescriptorPtr> &input, const ModelParams &params)
-{
-    // Check for models correctness
-    if (input.empty())
-        throw std::runtime_error("No model returned by loader");
-
-    for (auto &md : input)
-    {
-        if (!md)
-            throw std::runtime_error("No model returned by loader");
-    }
-
-    // Update loaded model with loader properties (so we can have the
-    // information with which it was loaded)
-
-    for (auto &md : input)
-    {
-        *md = params;
-        addModel(md);
-    }
+    return needsCommit;
 }
 
-void Scene::visitModels(const std::function<void(Model &)> &functor)
+SceneModelManager &Scene::getModelManager() noexcept
```

Comment:
Just personal preference but I would call these getters getModels(), getClippingGeometries(), getLights(), etc, just for conciseness.